### PR TITLE
ENH (sugaring): -c | --cfg-proc option for rev-create to specify cfg_ procedure(s) to run

### PR DESCRIPTION
**Pull request by yarikoptic**
ATM there is no easy way to specify procedures to run on a freshly created dataset. But in my case it is a common use case to quickly create datasets where I know what type they should be - should not
place text files to git, or might be BIDS datasets, etc.  So it would be nice to be able to succinctly specify procedure(s) to be ran on the new dataset as to configure it.

Alternative implementation could be adding  --proc-post to the rev-create command itself. Pros then would be:
  - consistent with --proc-post for datalad command itself
  - could then run any procedure, not only cfg_ one

Cons:
  - would be notably more long/tedius to type - i.e.

        rev-create --proc-post cfg_text2git

    vs

        rev-create -c text2git

I also added a `lgr.info` call within `run_procedure` to signal what procedure is actually about to be ran.  Although might be considered a bit oxymoronic in the case of direct `run_procedure` invocation, makes sense everywhere else procedures could be invoked.

So now it would look like
```shell
$> datalad rev-create -c text2git -f /tmp/newds2        
[INFO   ] Creating a new annex repo at /tmp/newds2 
create(ok): /tmp/newds2 (dataset)                                                              
[INFO   ] Running procedure cfg_text2git 
[INFO   ] == Command start (output follows) ===== 
[INFO   ] == Command exit (modification check follows) ===== 
```

With something like this I think I wouldn't miss any other option of current `create` too much happen we just decide to replace `create` with `rev-create`.

TODOs:
- [ ] Document special purpose/treatment of `cfg_` procedures
- [ ] Decide on `cfg_` vs `setup_`?

### CODE REVIEW

File: datalad/interface/run_procedure.py [hunk 1]
```diff
@@ -451,6 +451,7 @@ def __call__(
         lgr.debug('Attempt to run procedure {} as: {}'.format(
             name,
             cmd))
+        lgr.info("Running procedure %s", name)
```
Comment (mih) [inline datalad/interface/run_procedure.py]: Maybe "upgrade" the debug message on the line before? Two log messages with almost identical content seem unnecessary.

**Comment by kyleam** [comment]
Should we decide the fate of command hooks (#3264) before going forward with this?

File: datalad/interface/run_procedure.py (hunk 1 above)
Comment (yarikoptic) [reply datalad/interface/run_procedure.py]: Indeed! Good catch - I was blind.  Making it 
```python
        lgr.info('Running procedure %s', name)
        lgr.debug('Procedure command is %r', cmd)
```
this way we can see gory details of the command for debugging purposes (like apparently there is some trailing space ATM, does not matter but might come critical, who knows).

BTW looking at the 
```python
        lgr.debug('Attempt to run procedure {} as: {}'.format(
            name,
            cmd))
```
it is imho better to delay any string formatting until (and if) actually needed, that is why I use `%` form, and would do
```python
        lgr.debug('Attempt to run procedure %s as: %s', name, cmd)
```
I didn't time the effects, but assume that it should spare a cycle or two ;-)

**Comment by yarikoptic** [comment]
> Should we decide the fate of command hooks (#3264) before going forward with this?

as [I have expressed myself](https://github.com/datalad/datalad/issues/3264#issuecomment-475870474) hooks specification doesn't allow for quick/concise option to achieve trivial dataset customization upon `create`.  Although might be more generic and useful (although besides this `create` use case I do not have immediate others yet), it would probably largely be complimentary. May be even this sugaring functionality could be refactored later to (ab)use hooks for some reason instead of invoking `run_procedure` directly.

Overall, I do not think there is need to decide on hooks before starting to provide this convenience `-c` option.  Or am I missing something?

**Comment by yarikoptic** [comment]
My concern with this PR is necessity to agree on the `cfg_` prefix for this purpose and us already having at least two - `cfg_` and `setup_` which IMHO are the same thing here semantically:

```shell
$> datalad run-procedure --discover                                        
setup_bids_dataset (datalad_hirni/resources/procedures/setup_bids_dataset.py) [python_script]
setup_hirni_dataset (datalad_hirni/resources/procedures/setup_hirni_dataset.py) [python_script]
cfg_metadatatypes (../datalad-master/datalad/resources/procedures/cfg_metadatatypes.py) [python_script]
cfg_text2git (../datalad-master/datalad/resources/procedures/cfg_text2git.py) [python_script]
setup_yoda_dataset (../datalad-master/datalad/resources/procedures/setup_yoda_dataset.py) [python_script]
```
Do you think we could rename `setup_` to `cfg_` or there is some principled difference?

**Comment by kyleam** [comment]
[quoting out of order]

@yarikoptic:

> Overall, I do not think there is need to decide on hooks before starting to provide this convenience `-c` option.  Or am I missing something?

I think you're right that this shouldn't be held up by command hooks.
If we end up adding hooks (and specifically a `create-done` one), this
should be easy enough to map onto that without any API change.

> as [I have expressed myself](https://github.com/datalad/datalad/issues/3264#issuecomment-475870474) hooks specification doesn't allow for quick/concise option to achieve trivial dataset customization upon `create`.

My question wasn't implying that the syntax there was an alternative to
sugaring.

> Although might be more generic and useful (although besides this
> `create` use case I do not have immediate others yet), it would
> probably largely be complimentary.

AFAICT it calls `run_procedure` where the `create-done` hook would,
without being hooked up to the configuration and being restricted to
procedures "cfg_*".  So...

> May be even this sugaring functionality could be refactored later to (ab)use hooks for some reason instead of invoking `run_procedure` directly.

I'd guess that, if we do add hooks, we should refactor this to work with
the `run_procedure` loop for the `create-done` spot.

**Comment by mih** [comment]
I am fine with `cfg_` and adjusted accordingly: https://github.com/datalad/datalad-neuroimaging/pull/60

**Comment by mih** [comment]
> TODO: Document special purpose/treatment of cfg_ procedures

What do you envision here, it seem done already?

File: datalad/interface/run_procedure.py (hunk 1 above)
Comment (mih) [reply datalad/interface/run_procedure.py]: I pushed such a modification.

**Comment by mih** [comment]
@yarikoptic Pushed a small test too

**Comment by yarikoptic** [comment]
Thank you @mih for all the tune ups!  I am ok to merge it as is or we could may be discuss the issue about possible need of allowing for parametrization which I initially brough up in https://github.com/datalad/datalad-neuroimaging/pull/60#issuecomment-487283125 and may be we better decide upon right away to not require changing API later on:

... if we figure out how to "parametrize" additional actions/parametrizations scalably/flexibly.  E.g. here `bids_derivative` is probably just a flavor of `bids`, so we could may be better encode/interface parameters of the procedures? i.e. I want `bids` which supports `derivatives` specification and may be some other specifics which might be orthogonal to being bids raw or derivative (e.g. multisession vs single session) and thus ideally be in a single command and not elevating one aspect (raw-vs-derivative, or single-vs-multisession) into procedure name.
In the case of `-c` we probably then should follow (for consistency) what you have already for --proc-pre/post: `--proc-pre <PROCEDURE NAME> [ARGS ...]`: `-c bids derivative` or `-c bids flavor=derivative`, although I dislike those open ended options and preferred some encoding within the name (`-c bids[derivative]` or `-c bids[flavor=derivative]`).

What do you think?

**Comment by mih** [comment]
I think this is good for now. Any parameterization syntax should be discussed and implemented at a more global scope, as `run_procedure()`, the hooks, and storage of such specs in the config would all benefit from this.

**Comment by yarikoptic** [comment]
Fwiw run_procedure already has one in place AFAIK, that is what I thought to follow. Similar treatment is established by crawl-init

**Pull request merged**